src/tanks/Tancik.java

```diff
@@ -0,0 +1,49 @@
+package tanks;
+
+import java.util.Vector;
+
+import robocode.AdvancedRobot;
+
+public class tancik {
+
+    public class Tancik extends AdvancedRobot{
+
+        public Vector<Gene> genes;
+        public Tancik(Vector<Gene> genes) {
+            this.genes = genes;
+        }
+
+        public void run() {
+
+            for (Gene gene : genes) {
```

Comment:
Takto budeme prochazet vsechny geny, ale jak jsme se domluvili ve studentariu, geny v chromozomu bychom chteli mit rozdelene. Cast pujde do `run()` a zbyle casti do zbylych zvolenych metod jako treba `OnScannedRobot`. Tedy navrhuju do runu dat jen predem vybranou cast genu z `genes`.
```suggestion
            for (Gene gene : genesForRun) {
```